src/paas_charm/charm_state.py

```diff
@@ -89,6 +89,7 @@ def from_charm(  # pylint: disable=too-many-arguments
         s3_connection_info: dict[str, str] | None = None,
         saml_relation_data: typing.MutableMapping[str, str] | None = None,
         rabbitmq_uri: str | None = None,
+        tracing_relation_data: "TempoParameters | None" = None,
```

Comment:
I think the string is needed here

Reply:
Moved the classes to the top of file and removed ".